flink-connector-vitess-cdc/src/test/java/com/vervetica/cdc/connectors/vitess/table/VitessConnectorITCase.java

```diff
@@ -0,0 +1,271 @@
+                                //                        + "    timestamp_c TIMESTAMP(0),\n"
+                                //                        + "    file_uuid BYTES,\n"
+                                + "    primary key (`id`) not enforced"
+                                + ") WITH ("
+                                + " 'connector' = 'vitess-cdc',"
+                                + " 'tablet-type' = 'MASTER',"
+                                + " 'hostname' = '%s',"
+                                + " 'port' = '%s',"
+                                + " 'vtctl.hostname' = '%s',"
+                                + " 'vtctl.port' = '%s',"
+                                + " 'keyspace' = '%s',"
+                                + " 'table-name' = '%s'"
+                                + ")",
+                        VITESS_CONTAINER.getHost(),
+                        VITESS_CONTAINER.getGrpcPort(),
+                        VITESS_CONTAINER.getHost(),
+                        VITESS_CONTAINER.getVtctldGrpcPort(),
+                        VITESS_CONTAINER.getKeyspace(),
+                        "test.full_types");
+        tEnv.executeSql(sourceDDL);
+
+        // async submit job
+        TableResult result = tEnv.executeSql("SELECT * FROM full_types");
+
+        // Vitess source doesn't read snapshot data. Source will be empty at first.
+        // There's no way knowing if it's started, using sleep here.
+        Thread.sleep(10000);
+
+        try (Connection connection = getJdbcConnection();
+                Statement statement = connection.createStatement()) {
+            statement.execute(
+                    "INSERT INTO test.full_types VALUES (\n"
+                            + "    DEFAULT, 127, 255, 32767, 65535, 2147483647, 4294967295, 2147483647, 9223372036854775807,\n"
+                            + "    'Hello World', 'abc', 123.102, 404.4443, 123.4567, 345.6, true);");
+            statement.execute("UPDATE test.full_types SET varchar_c = 'Bye World' WHERE id=1;");
+        }
+
+        waitForSnapshotStarted(result.collect());
+
+        List<String> expected =
+                Arrays.asList(
+                        "+I[1, 127, 255, 32767, 65535, 2147483647, 4294967295, 2147483647, 9223372036854775807, Hello World, abc, 123.102, 404.4443, 123.4567, 346, true]",
+                        "-U[1, 127, 255, 32767, 65535, 2147483647, 4294967295, 2147483647, 9223372036854775807, Hello World, abc, 123.102, 404.4443, 123.4567, 346, true]",
+                        "+U[1, 127, 255, 32767, 65535, 2147483647, 4294967295, 2147483647, 9223372036854775807, Bye World, abc, 123.102, 404.4443, 123.4567, 346, true]");
+
+        List<String> actual = fetchRows(result.collect(), expected.size());
+        assertEquals(expected, actual);
+        result.getJobClient().get().cancel().get();
+    }
+
+    private static List<String> fetchRows(Iterator<Row> iter, int size) {
```

Comment:
These codes are reused in many places. Could you help to add a util class in the common package?

Reply:
it would require refactoring of other modules not only vitess, I would prefer to do this in a separate pull request